app.py

```diff
@@ -0,0 +1,295 @@
+from flask import Flask, render_template, redirect, url_for, flash, abort
+from flask_sqlalchemy import SQLAlchemy
+from flask_wtf import FlaskForm
+from wtforms import StringField, PasswordField, SubmitField, FloatField, DecimalField
+from wtforms.validators import DataRequired, Email, EqualTo
+import os
+from datetime import datetime
+from flask_bcrypt import Bcrypt
+from flask_login import (
+    LoginManager,
+    UserMixin,
+    current_user,
+    logout_user,
+    login_user,
+    login_required,
+)
+
+app = Flask(__name__)
+basedir = os.path.abspath(os.path.dirname(__file__))
+print(basedir)
+app.config["SQLALCHEMY_DATABASE_URI"] = "sqlite:///" + os.path.join(basedir, "db.db")
+app.config["SECRET_KEY"] = "your_secret_key_here"
+db = SQLAlchemy(app)
+
+# Encrypting and Decrypting password
+bcrypt = Bcrypt(app)
+
+# Authentication
+login_manager = LoginManager()
+login_manager.init_app(app)
+login_manager.login_view = "login"
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User.query.get(int(user_id))
+
+
+# ===== MODELS ========================================================================
+
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    email = db.Column(db.String(50), unique=True, nullable=False)
+    name = db.Column(db.String(50), unique=True, nullable=False)
+    password = db.Column(db.String(200), nullable=False)
+    groups = db.relationship(
+        "Group", secondary="user_group", backref=db.backref("members", lazy="dynamic")
+    )
+
+
+# Association table between User and Group
+user_group = db.Table(
+    "user_group",
+    db.Column("user_id", db.Integer, db.ForeignKey("user.id")),
+    db.Column("group_id", db.Integer, db.ForeignKey("group.id")),
+)
+
+
+class Group(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(50), unique=True, nullable=False)
+    bills = db.relationship("Bill", backref="group", lazy=True)
+
+
+class Bill(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(50), nullable=False)
+    amount = db.Column(db.Float, nullable=False)
+    group_id = db.Column(db.Integer, db.ForeignKey("group.id"))
+
+
+# ===== ROUTES ========================================================================
+
+
+@app.route("/", methods=["GET", "POST"])
+def route_index():
+    db.create_all()
```

Comment:
Pačiam / kelyje naudoti db.create_all() komandos negalima, nes kiekvieną kartą, kai einama į pagrindinį kelio tašką, ši komanda yra iš naujo vykdoma be reikalo

Reply:
Supratau, netyciukas, pasalinsiu.